File: test/tests/System/Monitor.js
describe('Monitor', function() {
    let ctx;

    before(function() {
        ctx = new randar.GraphicsContext();
    });

    it('presents geometry', function() {
        this.timeout(10000);
        let geo = randar.generate.sphere(0.5, 16, 16);

        let monitor = ctx.monitor(geo);
        monitor.window().fps(24);

        for (let i = 0; i < 48; i++) {
            monitor.window().camera().move(randar.vector(0, 0, 0.05));
            monitor.present();
        }

        monitor.window().close();
    });
});

File: modules/engine/src/System/Monitor.cpp
#include <randar/System/Monitor.hpp>

// Constructor.
randar::Monitor::Monitor(randar::GraphicsContext& context, randar::Geometry& geometry)
: randar::GraphicsContextResource(&context),
  geo(&geometry)
{
    this->win = &this->ctx->window(800, 600);
    this->win->camera().projection();
    this->win->camera().move(Vector3(0, 0, -10));
    this->win->camera().target(Vector3(0, 0, 0));
}

// Assignment.
randar::Monitor::Monitor(randar::Monitor&& other)
: randar::GraphicsContextResource(&other.context())
{
    this->win = other.win;
    other.win = nullptr;

    this->geo = other.geo;
    other.geo = nullptr;
}

// Destructor.
randar::Monitor::~Monitor()
{
    delete this->win;
}

// Presents the monitor to the screen.
void randar::Monitor::present()
{
    this->win->clear();

    if (this->geo) {
        this->win->draw(*this->geo);
    }

    this->win->present();
}
